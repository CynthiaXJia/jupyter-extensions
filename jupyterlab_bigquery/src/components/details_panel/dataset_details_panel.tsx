import * as React from 'react';

import {
  DatasetDetailsService,
  DatasetDetails,
  DatasetDetailsObject,
} from './service/list_dataset_details';
import { Header } from '../shared/header';
import LoadingPanel from '../loading_panel';
import { DetailsPanel } from './details_panel';
import { stylesheet } from 'typestyle';
import { formatDate, formatMs } from '../../utils/formatters';
import { BASE_FONT } from 'gcp_jupyterlab_shared';

export const localStyles = stylesheet({
  body: {
    marginRight: '24px',
    marginLeft: '24px',
    flex: 1,
    minHeight: 0,
    overflowY: 'auto',
    overflowX: 'hidden',
  },
  container: {
    height: '100%',
    display: 'flex',
    flexDirection: 'column',
    ...BASE_FONT,
  },
});

interface Props {
  datasetDetailsService: DatasetDetailsService;
  isVisible: boolean;
  dataset_id: string;
}

interface State {
  hasLoaded: boolean;
  isLoading: boolean;
  details: DatasetDetails;
  rows: DetailRow[];
}

interface DetailRow {
  name: string;
  value: string;
}

export function prepDatasetDetailsRows(details: DatasetDetailsObject) {
  return [
    { name: 'Dataset ID', value: details.id },
    { name: 'Created', value: formatDate(details.date_created) },
    {
      name: 'Default table expiration',
      value: details.default_expiration
        ? formatMs(details.default_expiration)
        : 'Never',
    },
    {
      name: 'Last modified',
      value: formatDate(details.last_modified),
    },
    {
      name: 'Data location',
      value: details.location ? details.location : 'None',
    },
  ];
}

export default class DatasetDetailsPanel extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props);
    this.state = {
      hasLoaded: false,
      isLoading: false,
      details: { details: {} } as DatasetDetails,
      rows: [],
    };
  }

  componentDidUpdate(prevProps: Props) {
    const isFirstLoad =
      !(this.state.hasLoaded || prevProps.isVisible) && this.props.isVisible;
    if (isFirstLoad) {
      this.getDetails();
    }
  }

  private async getDetails() {
    try {
      this.setState({ isLoading: true });
      const details = await this.props.datasetDetailsService.listDatasetDetails(
        this.props.dataset_id
      );

      const detailsObj = details.details;
      const rows = prepDatasetDetailsRows(detailsObj);

      this.setState({ hasLoaded: true, details, rows });
    } catch (err) {
      console.warn('Error retrieving dataset details', err);
    } finally {
      this.setState({ isLoading: false });
    }
  }

  render() {
    if (this.state.isLoading) {
      return <LoadingPanel />;
    } else {
      return (
        <div className={localStyles.container}>
          <Header>{this.props.dataset_id}</Header>
          <div className={localStyles.body}>
            <DetailsPanel
              details={this.state.details.details}
              rows={this.state.rows}
              detailsType="DATASET"
            />
          </div>
        </div>
      );
    }
  }
}
