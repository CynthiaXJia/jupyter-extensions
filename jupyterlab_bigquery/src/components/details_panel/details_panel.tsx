import * as React from 'react';

import {
  Grid,
  Chip,
  Table,
  TableBody,
  TableCell,
  TableHead,
  TableRow,
  withStyles,
} from '@material-ui/core';
import { stylesheet } from 'typestyle';
import { SchemaField } from './service/list_table_details';

export const localStyles = stylesheet({
  header: {
    borderBottom: 'var(--jp-border-width) solid var(--jp-border-color2)',
    fontSize: '18px',
    margin: 0,
    padding: '8px 12px 8px 24px',
  },
  title: {
    fontSize: '16px',
    marginBottom: '8px',
  },
  panel: {
    backgroundColor: 'white',
    height: '100%',
    display: 'flex',
    flexDirection: 'column',
  },
  detailsBody: {
    fontSize: '13px',
    flex: 1,
    minHeight: 0,
    display: 'flex',
    flexDirection: 'column',
    marginTop: '24px',
  },
  labelContainer: {
    display: 'flex',
    flexWrap: 'wrap',
    '& > *': {
      margin: '4px',
    },
  },
  rowTitle: {
    width: '200px',
  },
  row: {
    display: 'flex',
    padding: '6px',
  },
  scrollable: {
    flex: 1,
    minHeight: 0,
    overflow: 'auto',
  },
});

const TableHeadCell = withStyles({
  root: {
    backgroundColor: '#f0f0f0',
  },
})(TableCell);

const getStripedStyle = index => {
  return { background: index % 2 ? 'white' : '#fafafa' };
};

const formatFieldName = name => {
  if (name.includes('.')) {
    const child = name.substr(name.lastIndexOf('.') + 1);
    const parents = name.substr(0, name.lastIndexOf('.') + 1);
    return (
      <div>
        {parents} <b>{child}</b>
      </div>
    );
  } else {
    return <b>{name}</b>;
  }
};

interface SharedDetails {
  id: string;
  description: string;
  labels: string[];
  name: string;
  schema?: SchemaField[];
}

interface Props {
  details: SharedDetails;
  rows: any[];
  // TODO(cxjia): figure out a shared typing for these rows
  detailsType: string;
}

export const DetailsPanel: React.SFC<Props> = props => {
  const { details, rows, detailsType } = props;

  return (
    <div className={localStyles.panel}>
      <div className={localStyles.detailsBody}>
        <Grid container spacing={4}>
          <Grid item xs={6}>
            <div className={localStyles.title}>Description</div>
            <div>{details.description ? details.description : 'None'}</div>
          </Grid>

          <Grid item xs={6}>
            <div>
              <div className={localStyles.title}>Labels</div>
              {details.labels ? (
                <div className={localStyles.labelContainer}>
                  {details.labels.map((value, index) => {
                    return <Chip size="small" key={index} label={value} />;
                  })}
                </div>
              ) : (
                'None'
              )}
            </div>
          </Grid>

          <Grid item xs={12}>
            <div className={localStyles.title}>
              {detailsType === 'table' ? 'Table' : 'Dataset'} info
            </div>
            {rows.map((row, index) => (
              <div
                key={index}
                className={localStyles.row}
                style={{ ...getStripedStyle(index) }}
              >
                <div className={localStyles.rowTitle}>
                  <b>{row.name}</b>
                </div>
                <div>{row.value}</div>
              </div>
            ))}
          </Grid>
        </Grid>

        {detailsType === 'table' && (
          <div className={localStyles.title} style={{ marginTop: '32px' }}>
            Schema
          </div>
        )}

        {detailsType === 'table' && (
          <div className={localStyles.scrollable}>
            {details.schema && details.schema.length > 0 ? (
              <Table
                size="small"
                style={{ width: 'auto', tableLayout: 'auto' }}
              >
                <TableHead>
                  <TableRow>
                    <TableHeadCell>Field name</TableHeadCell>
                    <TableHeadCell>Type</TableHeadCell>
                    <TableHeadCell>Mode</TableHeadCell>
                    <TableHeadCell>Description</TableHeadCell>
                  </TableRow>
                </TableHead>
                <TableBody>
                  {details.schema.map((field, index) => {
                    return (
                      <TableRow key={`schema_row_${index}`}>
                        <TableCell>{formatFieldName(field.name)}</TableCell>
                        <TableCell>{field.type}</TableCell>
                        <TableCell>{field.mode}</TableCell>
                        <TableCell>{field.description ?? ''}</TableCell>
                      </TableRow>
                    );
                  })}
                </TableBody>
              </Table>
            ) : (
              'Table does not have a schema.'
            )}
          </div>
        )}
      </div>
    </div>
  );
};
