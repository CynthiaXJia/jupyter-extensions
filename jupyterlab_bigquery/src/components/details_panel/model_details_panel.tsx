import * as React from 'react';
import {
  ModelDetailsService,
  ModelDetails,
} from './service/list_model_details';
import LoadingPanel from '../loading_panel';
import { DetailsPanel } from './details_panel';

interface Props {
  modelDetailsService: ModelDetailsService;
  isVisible: boolean;
  tableId: string;
}

interface State {
  hasLoaded: boolean;
  isLoading: boolean;
  details: ModelDetails;
  rows: DetailRow[];
}

interface DetailRow {
  name: string;
  value: string | number;
}

export default class ModelDetailsPanel extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props);
    this.state = {
      hasLoaded: false,
      isLoading: false,
      details: { details: {} } as ModelDetails,
      rows: [],
    };
  }

  componentDidUpdate(prevProps: Props) {
    const isFirstLoad =
      !(this.state.hasLoaded || prevProps.isVisible) && this.props.isVisible;
    if (isFirstLoad) {
      this.getDetails();
    }
  }

  private async getDetails() {
    try {
      this.setState({ isLoading: true });
      const details = await this.props.modelDetailsService.listModelDetails(
        this.props.tableId
      );

      const detailsObj = details.details;
      const rows = [
        { name: 'Model ID', value: detailsObj.id },
        { name: 'Date created', value: detailsObj.date_created },
        {
          name: 'Model expiration',
          value: detailsObj.expires ? detailsObj.expires : 'Never',
        },
        {
          name: 'Date modified',
          value: detailsObj.last_modified,
        },
        {
          name: 'Data location',
          value: detailsObj.location ? detailsObj.location : 'None',
        },
        {
          name: 'Model type',
          value: detailsObj.model_type,
        },
        // {
        //     name: 'Loss type',
        //     value: detailsObj.loss_type
        // }
      ];
      this.setState({ hasLoaded: true, details, rows });
    } catch (err) {
      console.warn('Error retrieving table details', err);
    } finally {
      this.setState({ isLoading: false });
    }
  }

  render() {
    if (this.state.isLoading) {
      return <LoadingPanel />;
    } else {
      return (
        <DetailsPanel
          details={this.state.details.details}
          rows={this.state.rows}
          detailsType="MODEL"
        />
      );
    }
  }
}
