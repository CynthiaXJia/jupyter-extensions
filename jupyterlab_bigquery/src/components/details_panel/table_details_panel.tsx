import * as React from 'react';
import {
  TableDetailsService,
  TableDetails,
  TableDetailsObject,
} from './service/list_table_details';
import LoadingPanel from '../loading_panel';
import { DetailsPanel } from './details_panel';
import { formatDate, formatBytes } from '../../utils/formatters';

interface Props {
  tableDetailsService: TableDetailsService;
  isVisible: boolean;
  tableId: string;
}

interface State {
  hasLoaded: boolean;
  isLoading: boolean;
  details: TableDetails;
  rows: DetailRow[];
}

interface DetailRow {
  name: string;
  value: string | number;
}

export function prepTableDetailsRows(details: TableDetailsObject) {
  return [
    { name: 'Table ID', value: details.id },
    {
      name: 'Table size',
      value: formatBytes(details.num_bytes),
    },
    {
      name: 'Number of rows',
      value: details.num_rows.toLocaleString(),
    },
    { name: 'Created', value: formatDate(details.date_created) },
    {
      name: 'Table expiration',
      value: details.expires ? formatDate(details.expires) : 'Never',
    },
    {
      name: 'Last modified',
      value: formatDate(details.last_modified),
    },
    {
      name: 'Data location',
      value: details.location ? details.location : 'None',
    },
  ];
}

export default class TableDetailsPanel extends React.Component<Props, State> {
  private mounted = false;
  constructor(props: Props) {
    super(props);
    this.state = {
      hasLoaded: false,
      isLoading: false,
      details: { details: {} } as TableDetails,
      rows: [],
    };
  }

  componentDidMount() {
    this.mounted = true;
  }

  componentWillUnmount() {
    this.mounted = false;
  }

  componentDidUpdate(prevProps: Props) {
    const isFirstLoad =
      !(this.state.hasLoaded || prevProps.isVisible) && this.props.isVisible;
    if (isFirstLoad) {
      this.getDetails();
    }
  }

  private async getDetails() {
    try {
      if (this.mounted) {
        this.setState({ isLoading: true });
      }

      const details = await this.props.tableDetailsService.listTableDetails(
        this.props.tableId
      );

      const detailsObj = details.details;
      const rows = prepTableDetailsRows(detailsObj);
      if (this.mounted) {
        this.setState({ hasLoaded: true, details, rows });
      }
    } catch (err) {
      console.warn('Error retrieving table details', err);
    } finally {
      if (this.mounted) {
        this.setState({ isLoading: false });
      }
    }
  }

  render() {
    if (this.state.isLoading) {
      return <LoadingPanel />;
    } else {
      return (
        <DetailsPanel
          details={this.state.details.details}
          rows={this.state.rows}
          detailsType="TABLE"
        />
      );
    }
  }
}
