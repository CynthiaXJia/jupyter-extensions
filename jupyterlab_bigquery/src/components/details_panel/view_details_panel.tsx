import * as React from 'react';

import { ViewDetailsService, ViewDetails } from './service/list_view_details';
import { Header } from '../shared/header';
import LoadingPanel from '../loading_panel';
import { DetailsPanel } from './details_panel';
import { stylesheet } from 'typestyle';

const localStyles = stylesheet({
  body: {
    marginBottom: '24px',
    marginRight: '24px',
    marginLeft: '24px',
    height: '100%',
    overflowY: 'auto',
  },
});

interface Props {
  viewDetailsService: ViewDetailsService;
  isVisible: boolean;
  view_id: string;
  view_name: string;
}

interface State {
  hasLoaded: boolean;
  isLoading: boolean;
  details: ViewDetails;
  rows: DetailRow[];
}

interface DetailRow {
  name: string;
  value: string;
}

export default class ViewDetailsPanel extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props);
    this.state = {
      hasLoaded: false,
      isLoading: false,
      details: { details: {} } as ViewDetails,
      rows: [],
    };
  }

  componentDidUpdate(prevProps: Props) {
    const isFirstLoad =
      !(this.state.hasLoaded || prevProps.isVisible) && this.props.isVisible;
    if (isFirstLoad) {
      this.getDetails();
    }
  }

  formatMs(ms) {
    const days = ms / 86400000;
    return `${days} day${days > 1 ? 's' : ''} 0 hr`;
  }

  private async getDetails() {
    try {
      this.setState({ isLoading: true });
      const details = await this.props.viewDetailsService.listViewDetails(
        this.props.view_id
      );

      const detailsObj = details.details;
      const rows = [
        { name: 'View ID', value: detailsObj.id },
        { name: 'Created', value: detailsObj.date_created },
        { name: 'Last modified', value: detailsObj.last_modified },
        { name: 'View expiration', value: detailsObj.expires ?? 'Never' },
        { name: 'Use Legacy SQL', value: detailsObj.legacy_sql },
      ];

      this.setState({ hasLoaded: true, details, rows });
    } catch (err) {
      console.warn('Error retrieving view details', err);
    } finally {
      this.setState({ isLoading: false });
    }
  }

  render() {
    if (this.state.isLoading) {
      return <LoadingPanel />;
    } else {
      return (
        <div style={{ height: '100%' }}>
          <Header text={this.props.view_name} />
          <div className={localStyles.body}>
            <DetailsPanel
              details={this.state.details.details}
              rows={this.state.rows}
              detailsType="VIEW"
            />
          </div>
        </div>
      );
    }
  }
}
