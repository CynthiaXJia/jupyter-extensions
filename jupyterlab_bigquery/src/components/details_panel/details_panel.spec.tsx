import * as React from 'react';
import { shallow } from 'enzyme';
import { Settings } from 'luxon';

import { DetailsPanel } from './details_panel';
import { prepDatasetDetailsRows } from './dataset_details_panel';
import { prepTableDetailsRows } from './table_details_panel';
import { prepViewDetailsRows } from './view_details_panel';
import {
  prepModelDetailsRows,
  prepTrainingRunDetailsRows,
} from './model_details_panel';
import {
  MockDatasetDetailsFull,
  MockDatasetDetailsEmpty,
  MockTableDetailsFull,
  MockTableDetailsEmpty,
  MockViewDetailsFull,
  MockModelDetailsSingleRun,
  MockTrainingRunDetails,
} from './test_helpers';

describe('DetailsPanel', () => {
  beforeEach(() => {
    Settings.defaultZoneName = 'America/Los_Angeles';
  });

  afterEach(() => {
    Settings.defaultZoneName = 'local';
  });

  it('Renders for datasets with description and labels', () => {
    const mockDatasetDetails = MockDatasetDetailsFull;
    const mockRows = prepDatasetDetailsRows(mockDatasetDetails);
    const component = shallow(
      <DetailsPanel
        details={mockDatasetDetails}
        rows={mockRows}
        detailsType="DATASET"
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('Renders for datasets without description and labels', () => {
    const mockDatasetDetails = MockDatasetDetailsEmpty;
    const mockRows = prepDatasetDetailsRows(mockDatasetDetails);
    const component = shallow(
      <DetailsPanel
        details={mockDatasetDetails}
        rows={mockRows}
        detailsType="DATASET"
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('Renders for tables with all values populated', () => {
    const mockTableDetails = MockTableDetailsFull;
    const mockRows = prepTableDetailsRows(mockTableDetails);
    const component = shallow(
      <DetailsPanel
        details={mockTableDetails}
        rows={mockRows}
        detailsType="TABLE"
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('Renders for tables with some empty details', () => {
    const mockTableDetails = MockTableDetailsEmpty;
    const mockRows = prepTableDetailsRows(mockTableDetails);
    const component = shallow(
      <DetailsPanel
        details={mockTableDetails}
        rows={mockRows}
        detailsType="TABLE"
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('Renders for views', () => {
    const mockViewDetails = MockViewDetailsFull;
    const mockRows = prepViewDetailsRows(mockViewDetails);
    const component = shallow(
      <DetailsPanel
        details={mockViewDetails}
        rows={mockRows}
        detailsType="VIEW"
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('Renders a read-only editor for views', () => {
    const mockViewDetails = MockViewDetailsFull;
    const mockRows = prepViewDetailsRows(mockViewDetails);
    const component = shallow(
      <DetailsPanel
        details={mockViewDetails}
        rows={mockRows}
        detailsType="VIEW"
      />
    );
    expect(component.find('ReadOnlyEditor').exists()).toBeTruthy();
  });

  it('Renders for models', () => {
    const mockModelDetails = MockModelDetailsSingleRun;
    const mockRows = prepModelDetailsRows(mockModelDetails);
    const mockTrainingRows = prepTrainingRunDetailsRows(MockTrainingRunDetails);
    const component = shallow(
      <DetailsPanel
        details={mockModelDetails}
        rows={mockRows}
        trainingRows={mockTrainingRows}
        detailsType="MODEL"
      />
    );
    expect(component).toMatchSnapshot();
  });
});
