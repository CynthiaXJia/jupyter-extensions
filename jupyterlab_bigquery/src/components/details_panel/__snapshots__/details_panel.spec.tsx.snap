// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsPanel Renders for datasets with description and labels 1`] = `
<div
  className="panel_f1x07ye"
>
  <div
    className="detailsBody_fmw8kkq"
  >
    <WithStyles(ForwardRef(Grid))
      container={true}
      spacing={4}
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={6}
      >
        <div
          className="title_fvro9lr"
        >
          Description
        </div>
        <div>
          description of dataset
        </div>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={6}
      >
        <div>
          <div
            className="title_fvro9lr"
          >
            Labels
          </div>
          <div
            className="labelContainer_f14ovan"
          >
            <WithStyles(Component)
              darkmode={false}
              key="0"
              label="label: value"
              size="small"
            />
            <WithStyles(Component)
              darkmode={false}
              key="1"
              label="another label: another value"
              size="small"
            />
          </div>
        </div>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={12}
      >
        <div
          className="title_fvro9lr"
        >
          Dataset info
        </div>
        <Component
          rows={
            Array [
              Object {
                "name": "Dataset ID",
                "value": "projectId.datasetId",
              },
              Object {
                "name": "Created",
                "value": "Jun 15, 2020, 12:15:23 PM",
              },
              Object {
                "name": "Default table expiration",
                "value": "200 days 0 hr",
              },
              Object {
                "name": "Last modified",
                "value": "Jul 14, 2020, 9:52:02 AM",
              },
              Object {
                "name": "Data location",
                "value": "US",
              },
            ]
          }
        />
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </div>
</div>
`;

exports[`DetailsPanel Renders for datasets without description and labels 1`] = `
<div
  className="panel_f1x07ye"
>
  <div
    className="detailsBody_fmw8kkq"
  >
    <WithStyles(ForwardRef(Grid))
      container={true}
      spacing={4}
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={6}
      >
        <div
          className="title_fvro9lr"
        >
          Description
        </div>
        <div>
          None
        </div>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={6}
      >
        <div>
          <div
            className="title_fvro9lr"
          >
            Labels
          </div>
          None
        </div>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={12}
      >
        <div
          className="title_fvro9lr"
        >
          Dataset info
        </div>
        <Component
          rows={
            Array [
              Object {
                "name": "Dataset ID",
                "value": "projectId.datasetId",
              },
              Object {
                "name": "Created",
                "value": "Jun 15, 2020, 12:15:23 PM",
              },
              Object {
                "name": "Default table expiration",
                "value": "Never",
              },
              Object {
                "name": "Last modified",
                "value": "Jul 14, 2020, 9:52:02 AM",
              },
              Object {
                "name": "Data location",
                "value": "US",
              },
            ]
          }
        />
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </div>
</div>
`;

exports[`DetailsPanel Renders for models 1`] = `
<div
  className="panel_f1x07ye"
>
  <div
    className="detailsBody_fmw8kkq"
  >
    <WithStyles(ForwardRef(Grid))
      container={true}
      spacing={4}
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={6}
      >
        <div
          className="title_fvro9lr"
        >
          Description
        </div>
        <div>
          description of model
        </div>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={6}
      >
        <div>
          <div
            className="title_fvro9lr"
          >
            Labels
          </div>
          <div
            className="labelContainer_f14ovan"
          >
            <WithStyles(Component)
              darkmode={false}
              key="0"
              label="label: value"
              size="small"
            />
            <WithStyles(Component)
              darkmode={false}
              key="1"
              label="another label: another value"
              size="small"
            />
          </div>
        </div>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={12}
      >
        <div
          className="title_fvro9lr"
        >
          Model details
        </div>
        <Component
          rows={
            Array [
              Object {
                "name": "Model ID",
                "value": "projectId.datasetId.modelId",
              },
              Object {
                "name": "Date created",
                "value": "Jun 15, 2020, 12:15:23 PM",
              },
              Object {
                "name": "Model expiration",
                "value": "Aug 15, 2020, 9:52:02 AM",
              },
              Object {
                "name": "Date modified",
                "value": "Jul 14, 2020, 9:52:02 AM",
              },
              Object {
                "name": "Data location",
                "value": "US",
              },
              Object {
                "name": "Model type",
                "value": "LINEAR_REGRESSION",
              },
            ]
          }
        />
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={12}
      >
        <div
          className="title_fvro9lr"
        >
          Training options
        </div>
        <Component
          rows={
            Array [
              Object {
                "name": "Actual iterations",
                "value": 3,
              },
              Object {
                "name": "Data split column",
                "value": "column_name_1",
              },
              Object {
                "name": "Data split eval fraction",
                "value": 0.3,
              },
              Object {
                "name": "Data split method",
                "value": "RANDOM",
              },
              Object {
                "name": "Distance type",
                "value": "EUCLIDEAN",
              },
              Object {
                "name": "Early stop",
                "value": "False",
              },
              Object {
                "name": "Initial learn rate",
                "value": 0.1,
              },
              Object {
                "name": "Input label columns",
                "value": "['column1', 'column2']",
              },
              Object {
                "name": "Kmeans initialization column",
                "value": "column_name_2",
              },
              Object {
                "name": "Centroids initialization method",
                "value": "RANDOM",
              },
              Object {
                "name": "L1 regularization",
                "value": 0.5,
              },
              Object {
                "name": "L2 regularization",
                "value": 0.6,
              },
              Object {
                "name": "Label class weights",
                "value": "[0.2, 0.3]",
              },
              Object {
                "name": "Learn rate",
                "value": 0.7,
              },
              Object {
                "name": "Learn rate strategy",
                "value": "LINE_SEARCH",
              },
              Object {
                "name": "Loss type",
                "value": "MEAN_SQUARED_LOSS",
              },
              Object {
                "name": "Max allowed iterations",
                "value": 20,
              },
              Object {
                "name": "Min relative progress",
                "value": 0.4,
              },
              Object {
                "name": "Model uri",
                "value": "model.uri.string",
              },
              Object {
                "name": "Number of clusters",
                "value": 7,
              },
              Object {
                "name": "Optimization strategy",
                "value": "BATCH_GRADIENT_DESCENT",
              },
              Object {
                "name": "Warm start",
                "value": "True",
              },
            ]
          }
        />
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={12}
      >
        <div
          className="title_fvro9lr"
        >
          Label columns
        </div>
        <div>
          <Component
            schema={
              Array [
                Object {
                  "name": "predicted_field",
                  "type": "FLOAT",
                },
              ]
            }
          />
        </div>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={12}
      >
        <div
          className="title_fvro9lr"
        >
          Feature columns
        </div>
        <div>
          <Component
            schema={
              Array [
                Object {
                  "name": "feature_0",
                  "type": "INTEGER",
                },
                Object {
                  "name": "feature_1",
                  "type": "FLOAT",
                },
              ]
            }
          />
        </div>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </div>
</div>
`;

exports[`DetailsPanel Renders for tables with all values populated 1`] = `
<div
  className="panel_f1x07ye"
>
  <div
    className="detailsBody_fmw8kkq"
  >
    <WithStyles(ForwardRef(Grid))
      container={true}
      spacing={4}
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={6}
      >
        <div
          className="title_fvro9lr"
        >
          Description
        </div>
        <div>
          description of table
        </div>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={6}
      >
        <div>
          <div
            className="title_fvro9lr"
          >
            Labels
          </div>
          <div
            className="labelContainer_f14ovan"
          >
            <WithStyles(Component)
              darkmode={false}
              key="0"
              label="label: value"
              size="small"
            />
            <WithStyles(Component)
              darkmode={false}
              key="1"
              label="another label: another value"
              size="small"
            />
          </div>
        </div>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={12}
      >
        <div
          className="title_fvro9lr"
        >
          Table info
        </div>
        <Component
          rows={
            Array [
              Object {
                "name": "Table ID",
                "value": "projectId.datasetId.tableId",
              },
              Object {
                "name": "Table size",
                "value": "11.77 MB",
              },
              Object {
                "name": "Number of rows",
                "value": "123,456",
              },
              Object {
                "name": "Created",
                "value": "Jun 15, 2020, 12:15:23 PM",
              },
              Object {
                "name": "Table expiration",
                "value": "Aug 15, 2020, 9:52:02 AM",
              },
              Object {
                "name": "Last modified",
                "value": "Jul 14, 2020, 9:52:02 AM",
              },
              Object {
                "name": "Data location",
                "value": "US",
              },
            ]
          }
        />
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={12}
      >
        <div
          className="title_fvro9lr"
        >
          Schema
        </div>
        <Component
          schema={
            Array [
              Object {
                "description": "this field is a record",
                "mode": "NULLABLE",
                "name": "record",
                "type": "RECORD",
              },
              Object {
                "description": "this field is a string and a child",
                "mode": "NULLABLE",
                "name": "record.child",
                "type": "STRING",
              },
            ]
          }
        />
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </div>
</div>
`;

exports[`DetailsPanel Renders for tables with some empty details 1`] = `
<div
  className="panel_f1x07ye"
>
  <div
    className="detailsBody_fmw8kkq"
  >
    <WithStyles(ForwardRef(Grid))
      container={true}
      spacing={4}
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={6}
      >
        <div
          className="title_fvro9lr"
        >
          Description
        </div>
        <div>
          None
        </div>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={6}
      >
        <div>
          <div
            className="title_fvro9lr"
          >
            Labels
          </div>
          None
        </div>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={12}
      >
        <div
          className="title_fvro9lr"
        >
          Table info
        </div>
        <Component
          rows={
            Array [
              Object {
                "name": "Table ID",
                "value": "projectId.datasetId.tableId",
              },
              Object {
                "name": "Table size",
                "value": "11.77 MB",
              },
              Object {
                "name": "Number of rows",
                "value": "123,456",
              },
              Object {
                "name": "Created",
                "value": "Jun 15, 2020, 12:15:23 PM",
              },
              Object {
                "name": "Table expiration",
                "value": "Never",
              },
              Object {
                "name": "Last modified",
                "value": "Jul 14, 2020, 9:52:02 AM",
              },
              Object {
                "name": "Data location",
                "value": "US",
              },
            ]
          }
        />
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={12}
      >
        <div
          className="title_fvro9lr"
        >
          Schema
        </div>
        Table does not have a schema.
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </div>
</div>
`;

exports[`DetailsPanel Renders for views 1`] = `
<div
  className="panel_f1x07ye"
>
  <div
    className="detailsBody_fmw8kkq"
  >
    <WithStyles(ForwardRef(Grid))
      container={true}
      spacing={4}
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={6}
      >
        <div
          className="title_fvro9lr"
        >
          Description
        </div>
        <div>
          description of view
        </div>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={6}
      >
        <div>
          <div
            className="title_fvro9lr"
          >
            Labels
          </div>
          <div
            className="labelContainer_f14ovan"
          >
            <WithStyles(Component)
              darkmode={false}
              key="0"
              label="label: value"
              size="small"
            />
            <WithStyles(Component)
              darkmode={false}
              key="1"
              label="another label: another value"
              size="small"
            />
          </div>
        </div>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={12}
      >
        <div
          className="title_fvro9lr"
        >
          View info
        </div>
        <Component
          rows={
            Array [
              Object {
                "name": "View ID",
                "value": "projectId.datasetId.viewId",
              },
              Object {
                "name": "Created",
                "value": "Jun 15, 2020, 12:15:23 PM",
              },
              Object {
                "name": "Last modified",
                "value": "Jul 14, 2020, 9:52:02 AM",
              },
              Object {
                "name": "View expiration",
                "value": "Aug 15, 2020, 9:52:02 AM",
              },
              Object {
                "name": "Use Legacy SQL",
                "value": "true",
              },
            ]
          }
        />
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={12}
      >
        <div
          className="title_fvro9lr"
        >
          Schema
        </div>
        Table does not have a schema.
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={12}
      >
        <div
          className="title_fvro9lr"
        >
          Query
        </div>
        <ReadOnlyEditor
          query="SELECT * FROM \`projectId.datasetId.tableId\` LIMIT 100"
        />
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </div>
</div>
`;
